src/adapter/src/coord.rs

```diff
@@ -737,11 +772,122 @@ impl ExecuteContext {
             session,
             extra,
         } = self;
+        let reason = if extra.is_trivial() {
+            None
+        } else {
+            Some(match &result {
+                Ok(ok) => match ok {
+                    ExecuteResponse::CopyTo { resp, .. } => match resp.as_ref() {
+                        // NB [btv]: It's not clear that this combination
```

Comment:
`COPY (SELECT 1) TO STDOUT`? Although I'm confused why the other sending types panic at all, and also why this variant doesn't.